### PR TITLE
Fix JIT configs for aarch64

### PR DESCRIPTION
**Pull request by thinkharderdev**
# Which issue does this PR close?

Closes #.

 # Rationale for this change

Currently tests do not pass on Apple Silicone (or any non x86_64) platform. This PR adjusts JIT configs based on platform. 

Should no longer be needed once https://github.com/bytecodealliance/wasmtime/issues/2907 is closed. 

Disclaimer: I don't really know what I'm doing here and am mostly applying the workaround from https://github.com/bytecodealliance/wasmtime/issues/2735. 

# What changes are included in this PR?

Set the `is_pic` flag to `false` when creating `JIT` since `PLT` is not supported on non x86 platforms. 

# Are there any user-facing changes?

As far as I can tell, no

No

### CODE REVIEW

**Comment by alamb** [comment]
cc @yjshen

**Comment by matthewmturner** [comment]
I noticed this locally as well. Thx for fixing.

File: datafusion-jit/src/jit.rs [hunk 1]
```diff
@@ -51,6 +52,28 @@ impl Default for JIT {
...
     }
+
+    #[cfg(target_arch = "aarch64")]
+    fn default() -> Self {
+        let mut flag_builder = settings::builder();
+        // On at least AArch64, "colocated" calls use shorter-range relocations,
+        // which might not reach all definitions; we can't handle that here, so
+        // we require long-range relocation types.
+        flag_builder.set("use_colocated_libcalls", "false").unwrap();
+        flag_builder.set("is_pic", "false").unwrap();
```
Comment (houqp) [inline datafusion-jit/src/jit.rs:63]: perhaps worth linking to https://github.com/bytecodealliance/wasmtime/issues/2907 in the comment so we know to reenable pic when it's supported upstream.

**Review by houqp — APPROVED** [review]
@yjshen could you take a look at this?

File: datafusion-jit/src/jit.rs (hunk 1 above)
Comment (thinkharderdev) [reply datafusion-jit/src/jit.rs:63]: Good call

**Review by yjshen — APPROVED** [review]
LGTM

**Comment by yjshen** [comment]
Thanks again for fixing this. ❤️

**Review by Jimexist — APPROVED** [review]

**Pull request merged**